src/libraries/System.Text.RegularExpressions/tests/RegexParserTests.cs

```diff
@@ -578,258 +565,170 @@ static RegexParserTests()
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
-(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((""", RegexOptions.None, RegexParseError.NotEnoughParentheses)]
-        // Scan control
-        [InlineData("(cat)(\\c\0*)(dog)", RegexOptions.None, RegexParseError.UnrecognizedControl)]
-        [InlineData(@"(cat)(\c\[*)(dog)", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        // Invalid grouping constructs
-        [InlineData("(?<", RegexOptions.None, RegexParseError.UnrecognizedGrouping)]
-        [InlineData("(?>-", RegexOptions.None, RegexParseError.NotEnoughParentheses)]
-        // Testgroup with options
-        [InlineData("())", RegexOptions.None, RegexParseError.TooManyParentheses)]
-        [InlineData("[a-z-[aeiuo]", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        [InlineData("[a-z-[aeiuo", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        [InlineData("[a-z-[b]", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        [InlineData("[a-z-[b", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        [InlineData("[b-a]", RegexOptions.None, RegexParseError.ReversedCharRange)]
-        [InlineData(@"[a-c]{2,1}", RegexOptions.None, RegexParseError.IllegalRange)]
-        [InlineData(@"\d{2147483648}", RegexOptions.None, RegexParseError.CaptureGroupOutOfRange)]
-        [InlineData("[a-z-[b][", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        [InlineData("(?()|||||)", RegexOptions.None, RegexParseError.TooManyAlternates)]
-        [InlineData("[^]", RegexOptions.None, RegexParseError.UnterminatedBracket)]
-        public void Parse(string pattern, RegexOptions options, object errorObj)
+
+        public void Parse(string pattern, RegexOptions options, object errorObj, int offset = -1)
         {
             RegexParseError? error = (RegexParseError?)errorObj;
 
+            Assert.True(error == null || offset >= 0, "All errors must have offsets");
```

Comment:
Offsets are always greater than zero. The empty regex is valid, and a non-empty regex will give an error only after it parses the first character, i.e. `"*"` will have offset `1`.

Reply:
Ah. Yeah, I noticed some idiosyncracies in the offsets in general, but as far as I know they're always in a not-unreasonable place.

This line of code is simply making sure we added an offset in every test that expected an error. That is working. It can be tweaked at a future time.

Reply:
Ah, sorry, already fixed it ;)